### DISCUSSION TITLE
Client-side encryption for backups

### DISCUSSION THREAD

**Discussion by ArjonBu** [cloudnative-pg/cloudnative-pg#1172] [General]
Hello, 

Currently cloudnative-pg supports applying an encryption policy to the s3 objects which itself is encryption at rest. 
I don't see any option to have client-side encryption. Basically I want my databases to be encrypted in my cluster and then uploaded to the s3 endpoint. I do not want the cloud provider to be able to read my backups. 

Is this currently possible? [Rclone crypt remote](https://rclone.org/crypt/) is a good example on what I am looking for. 

Thanks in advamce

**Comment by gbartolini** [comment]
AFAIK Barman Cloud (which we rely on) does not support this. @theadamwright can you please confirm?

**Comment by sxd** [comment]
Turning this into a discussion since it's not an issue yet.

**Comment by theadamwright** [comment]
Encrypted backups have been discussed within the Barman project, but there are no firm plans at the moment. While there has been progress getting the necessary prerequisites in place, Transparent Data Encryption is moving slowly in the PostgreSQL community and requires relying on a 3rd party project or commercial vendor like EnterpriseDB. [Transparent Column Encryption](https://www.postgresql.org/message-id/89157929-c2b6-817b-6025-8e4b2d89d88f%40enterprisedb.com) is progressing the PostgreSQL community and appears to be on track for v16.
